toolchain/mfc/test/test.py

```diff
@@ -217,27 +213,31 @@ def _handle_case(case: TestCase, devices: typing.Set[int]):
 
 
 def handle_case(case: TestCase, devices: typing.Set[int]):
-    # pylint: disable=global-statement
-    global nFAIL
+    global nFAIL, nPASS, nSKIP
 
     nAttempts = 0
-
+    maxAttempts = ARG("max_attempts")
+    
+    if ARG("ci_mode"):
+        maxAttempts = 3
+    
     while True:
         nAttempts += 1
 
         try:
             _handle_case(case, devices)
+            nPASS += 1
         except Exception as exc:
-            if nAttempts < ARG("max_attempts"):
+            if nAttempts < maxAttempts:
                 cons.print(f"[bold yellow] Attempt {nAttempts}: Failed test {case.get_uuid()}. Retrying...[/bold yellow]")
                 continue
 
             nFAIL += 1
-
             cons.print(f"[bold red]Failed test {case} after {nAttempts} attempt(s).[/bold red]")
 
             if ARG("relentless"):
                 cons.print(f"{exc}")
+                nSKIP += 1
```

Comment:
this is unclear to me. it appears you are skipping every test case in `-r` mode? also at the moment it is unclear to me if we have any cases that are "skipped." a case I can think of is if a package is unavailable so then we skip the test for the case, but for now we have not had this (so may be worth removing?)

Reply:
Or I suppose this is saying that if we are in relentless mode, all failed cases are actually skipped cases? 

If so, this is not what I intended. If we have nfail > 0, we report this # at the end, then exit with a non-0 exit code.

Reply:
Addressed in most recent commit.